pkg/syncer/spec/spec_process.go

```diff
@@ -181,17 +199,29 @@ func (c *Controller) ensureDownstreamNamespaceExists(ctx context.Context, downst
 		})
 	}
 
-	// TODO(sttts): check that namespace exists in lister before using the client
-	if _, err := namespaces.Create(ctx, newNamespace, metav1.CreateOptions{}); err != nil {
-		// An already exists error is ok - it means something else beat us to creating the namespace.
-		if !k8serrors.IsAlreadyExists(err) {
-			// Any other error is not good, though.
-			// TODO bubble this up as a condition somewhere.
-			klog.Errorf("Error while creating namespace %q: %v", downstreamNamespace, err)
+	// Check if the namespace already exists, if not create it.
+	namespace, err := c.getDownstreamNamespace(newNamespace.GetName())
+	if apierrors.IsNotFound(err) {
+		// Not found, so let's create the namespace.
+		if _, err := namespaces.Create(ctx, newNamespace, metav1.CreateOptions{}); err != nil {
```

Comment:
handle AlreadyExist. You informer lookup will have latency to be correct.

Reply:
not sure what to do there, as if it already exists... we should loop again to make sure that the namespaceLocator matches what we want... so instead of repeating the logic there I just loop again.

Reply:
no, just assume your informer check few lines above was based on stale informers. The downstream namespace actually exists already.

Reply:
if we create another namespace with the same nsLocator, we will hit https://github.com/kcp-dev/kcp/pull/1309/files#diff-d0397d2587a97f7f7d490a7e6f83776c15aebd162c8ee61a4ebe19ad161777aaR135

Reply:
done